coff.ml

```diff
@@ -859,11 +946,17 @@ module Lib = struct
     let obj size name =
 (*      Printf.printf "-> %s (size %i)\n" name size;  *)
       let pos = pos_in ic in
-      if size > 18 && read_str ic pos 4 = "\000\000\255\255"
-      then imports := Import.read ic pos size :: !imports
```

Comment:
Was this an error before - having ascertained that the magic header was there, shouldn't have read the version, instead doesn't it get lumped onto the first symbol?

Reply:
Sort of. This check would treat ANON_OBJECT_HEADER_BIGOBJ as IMPORT_OBJECT_HEADER (i.e. an import library). I didn't try to feed bigobj files to the current version of flexlink but suppose it will fail somewhere.

Reply:
Ah, I see now (sorry, I could also have scoured the COFF spec, too) - what we had before was an assumption that the `\000\000\255\255` header implied strictly an object library whereas now we actually read the header properly. Gotcha!